client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/query/RemoteQuery.java

```diff
@@ -72,9 +81,20 @@ private void executeQuery() {
          validateNamedParameters();
 
          QueryOperation op = cache.getOperationsFactory().newQueryOperation(this, cache.getDataFormat());
-         QueryResponse response = await(op.execute());
-         totalResults = (int) response.getTotalResults();
-         results = unwrapResults(response.getProjectionSize(), response.getResults());
+         Object response = await(op.execute());
+         if (response instanceof QueryResponse) {
+            QueryResponse resp = (QueryResponse) response;
+            totalResults = (int) resp.getTotalResults();
+            results = unwrapResults(resp.getProjectionSize(), resp.getResults());
+         }
+         if (response instanceof JsonObject) {
```

Comment:
Why no else here?

Reply:
Because only two kinds of query response are possible. Actually it'd be better to extract an interface, let me fix this

Reply:
Sure, but the code would have allowed an unrecognised 3rd alternative to go unnoticed.

Reply:
indeed, I forgot to revisit this code :)